google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -338,34 +325,31 @@ def writes_for_update doc_path, data, update_time: nil
               raise ArgumentError, "data is required"
             end
 
+            write = Google::Cloud::Firestore::V1::Write.new(
+              update:           Google::Cloud::Firestore::V1::Document.new(name: doc_path),
+              update_mask:      Google::Cloud::Firestore::V1::DocumentMask.new,
+              current_document: Google::Cloud::Firestore::V1::Precondition.new(exists: true)
+            )
+
             if data.any? || delete_paths.any?
-              write = Google::Cloud::Firestore::V1::Write.new(
-                update: Google::Cloud::Firestore::V1::Document.new(
-                  name: doc_path,
-                  fields: hash_to_fields(data)),
-                update_mask: Google::Cloud::Firestore::V1::DocumentMask.new(
-                  field_paths: (field_paths).map(&:formatted_string).sort),
-                current_document: Google::Cloud::Firestore::V1::Precondition.new(
-                  exists: true)
-              )
+              htf = hash_to_fields data
+              htf.each_pair do |k, v|
+                write.update.fields[k] = v
+              end
+              write.update_mask.field_paths += field_paths.map(&:formatted_string).sort
+
               if update_time
-                write.current_document = \
-                  Google::Cloud::Firestore::V1::Precondition.new(
-                    update_time: time_to_timestamp(update_time))
+                write.current_document = Google::Cloud::Firestore::V1::Precondition.new(
+                  update_time: time_to_timestamp(update_time)
+                )
```

Comment:
@schmidt-sebastian Please review lines 328-344 to confirm that the `Write` looks correct for an update operation.

Reply:
I am pretty sure this is correct. It looks like it sends an empty update (with an existing but empty update mask) for a transfom-only update, which is what we want (otherwise it would delete the existing data).